### DISCUSSION TITLE
Stack images of varying size but only to a certain extent?

### DISCUSSION THREAD

**Discussion by s1-en** [ImageMagick/ImageMagick#1951] [General] [answered]
Is it possible to use `-append` to stack images of varying height (but same width) but contain the results to 800x1280px parts?

For example, I have these images

- image 1 (800x1000)
- image 2 (800x300)
- image 3 (800x600)

And after processing them, I should end up with the the images combined and cut up into parts

- 800x1280 (full image 1 and part of image 2)
- 800x620 (part of image 2 and full image 3)

I looked at the `-crop` flag but couldn't figure out how to get this specific behavior

**Comment by snibgo** [comment, marked as answer]
Simple: just append and crop. But you need "+repage" after each one:
`magick -size 800x1000 xc: -size 800x300 xc: -size 800x600 xc: -append +repage -crop 800x1280 +repage info:`

`xc:[0] XC 800x1280 16-bit sRGB 0.141u 0:00.037`
`xc:[1] XC 800x620 16-bit sRGB 0.094u 0:00.032`

**Reply by s1-en** [reply]
Do I need t specify each image's size? I'm planning to do a batch process on a bunch of images (more than 3) and was hoping ImageMagick would automatically detect it somehow

Also I read the details about `+repage` but I couldn't understand what exactly it does from how it's worded, English isn't my main language

Edit:  
Oh never mind, I completely forgot about wildcards! Thanks for the help!

**Comment by snibgo** [comment]
I showed you an example that creates three images and then does what you want. But you would use your own images, for example:

`magick img1.png img2.png img3.png -append +repage -crop 800x1280 +repage out.png`
